trulens_eval/trulens_eval/feedback/provider/base.py

```diff
@@ -339,20 +469,79 @@ def context_relevance_with_cot_reasons(
         Returns:
             float: A value between 0 and 1. 0 being "not relevant" and 1 being "relevant".
         """
-        system_prompt = prompts.CONTEXT_RELEVANCE_SYSTEM
+
         user_prompt = str.format(
             prompts.CONTEXT_RELEVANCE_USER, question=question, context=context
         )
         user_prompt = user_prompt.replace(
             "RELEVANCE:", prompts.COT_REASONS_TEMPLATE
         )
 
+        if criteria and output_space:
+            ContextRelevance.override_critera_and_output_space(
+                criteria, output_space
+            )
+
         return self.generate_score_and_reasons(
-            system_prompt=system_prompt,
+            system_prompt=ContextRelevance.system_prompt.template,
             user_prompt=user_prompt,
             temperature=temperature
         )
 
+    def context_relevance_verb_confidence(
+        self,
+        question: str,
+        context: str,
+        criteria: str = "",
+        output_space: str = "",
+        temperature: float = 0.0
+    ) -> Tuple[float | None, float | None]:
+        """
+        Uses chat completion model. A function that completes a
+        template to check the relevance of the context to the question.
+        Also uses chain of thought methodology and emits the reasons.
+
+        !!! example
+
+            ```python
+            from trulens_eval.app import App
+            context = App.select_context(rag_app)
+            feedback = (
+                Feedback(provider.context_relevance_with_cot_reasons)
+                .on_input()
+                .on(context)
+                .aggregate(np.mean)
+                )
+            ```
+
+        Args:
+            question (str): A question being asked.
+
+            context (str): Context related to the question.
+
+        Returns:
+            float: A value between 0 and 1. 0 being "not relevant" and 1 being "relevant".
+        """
+        if criteria and output_space:
+            ContextRelevance.override_critera_and_output_space(
+                criteria, output_space
+            )
+
+        try:
+            return self.generate_confidence_score(
+                verb_confidence_prompt=ContextRelevance.system_prompt.template +
+                ContextRelevance.verb_confidence_prompt.template,
+                user_prompt=str.format(
+                    prompts.CONTEXT_RELEVANCE_USER,
+                    question=question,
+                    context=context
+                ),
+                temperature=temperature
+            )
+        except ValueError as e:
+            logger.error(e)
+            return None, None
+
     def qs_relevance_with_cot_reasons(self, question: str,
```

Comment:
seems like a good time to actually deprecate qs_relevance so we don't have too many variations of context relevance in here.

Reply:
yep seems like a good time https://github.com/truera/trulens/pull/1299